### PR TITLE
#1359 location and location movements

### LINKED ISSUE #1359: Add Location, ItemBatchLocationMovement data types
## Is your feature request related to a problem? Please describe.

Need to add `Location` and `ItemBatchLocationMovement` as well as `batch` changes for location tracking

## Describe the solution you'd like

Add:

image

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
### BRANCHED FROM: #1358 

Fixes #1359 

## Change summary

- Adds `Location`
- Adds `ItemBatchLocationMovement`
- Adds `..Batch` field updates

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/database/DataTypes/LocationMovement.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import Realm from 'realm';
+
+export class LocationMovement extends Realm.Object {}
```
Comment (joshxg) [inline src/database/DataTypes/LocationMovement.js]: `ItemBatchLocationMovement`.. `LocationMovement`... i dunno!

File: src/database/DataTypes/ItemBatch.js [hunk 2]
```diff
@@ -128,6 +128,12 @@ ItemBatch.schema = {
     supplier: { type: 'Name', optional: true },
     donor: { type: 'Name', optional: true },
     transactionBatches: { type: 'list', objectType: 'TransactionBatch' },
+    location: { type: 'Location', optional: true },
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply src/database/DataTypes/ItemBatch.js]: I'm confused. why does a location have a type of 'Location'? Doesn't the name of the property already tell me that?

File: src/database/DataTypes/LocationMovement.js (hunk 1 above)
Comment (mark-prins) [reply src/database/DataTypes/LocationMovement.js]: I vote for `LocationMovement`

File: src/database/DataTypes/Location.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Location.schema = {
+  name: 'Location',
+  primaryKey: 'id',
+  properties: {
+    id: 'string',
+    description: { type: 'string', optional: true },
+    code: { type: 'string', optional: true },
+    locationType: { type: 'LocationType', optional: true },
+    locationMovements: {
```
Comment (mark-prins) [inline src/database/DataTypes/Location.js]: out of curiosity - does this map to something in mSupply?

**Comment by joshxg** [comment]
> I'm confused. why does a location have a type of 'Location'? Doesn't the name of the property already tell me that?

thats just saying the type for `ItemBatch.location` is a `Location`

> out of curiosity - does this map to something in mSupply?

Nah, these linking objects are essentially just short cut queries for one-to-many relationships! Equivalent to using an `orda` relationship

**Pull request merged**